smac/intensification/successive_halving.py

```diff
@@ -0,0 +1,475 @@
+    instance_order : str
+        how to order instances. Can be set to:
+        None - use as is given by the user
+        shuffle_once - shuffle once and use across all SH run (default)
+        shuffle - shuffle before every SH run
+    adaptive_capping_slackfactor : float
+        slack factor of adpative capping (factor * adpative cutoff)
+
+    Returns
+    --------
+    Configuration
+        new incumbent configuration
+    float
+        empirical performance of incumbent configuration
+    """
+
+    def __init__(self, tae_runner: ExecuteTARun,
+                 stats: Stats,
+                 traj_logger: TrajLogger,
+                 rng: np.random.RandomState,
+                 instances: typing.List[str],
+                 instance_specifics: typing.Mapping[str, np.ndarray] = None,
+                 cutoff: int = None,
+                 deterministic: bool = False,
+                 min_budget: float = None,
+                 max_budget: float = None,
+                 eta: float = 3,
+                 init_chal: int = None,
+                 run_obj_time: bool = True,
+                 n_seeds: int = None,
+                 instance_order: str = 'shuffle_once',
+                 adaptive_capping_slackfactor: float = 1.2,
+                 **kwargs):
+
+        super().__init__(tae_runner, stats, traj_logger, rng, instances,
+                         instance_specifics=instance_specifics, cutoff=cutoff,
+                         deterministic=deterministic, run_obj_time=run_obj_time,
+                         adaptive_capping_slackfactor=adaptive_capping_slackfactor)
+
+        self.logger = logging.getLogger(
+            self.__module__ + "." + self.__class__.__name__)
+
+        # INSTANCES
+
+        self.n_seeds = n_seeds if n_seeds else 1
+        self.instance_order = instance_order
+
+        # if instances are coming from Hyperband, skip the instance preprocessing section
+        # it is already taken care by Hyperband
+        if instances is not None and isinstance(instances[0], InstSeedKey):
+            self.instances = instances
+        else:
+            instances = [] if instances is None else instances
+            # removing duplicates in the user provided instances
+            self.instances = list(OrderedDict.fromkeys(instances))
+
+            # determine instance order
+            if self.instance_order == 'shuffle_once':
+                # randomize once
+                self.rs.shuffle(self.instances)
+
+            # set seed(s) for all SH runs
+            # - currently user gives the number of seeds to consider
+            if self.deterministic:
+                seeds = [0]
+            else:
+                seeds = self.rs.randint(low=0, high=MAXINT, size=self.n_seeds)
+            # storing instances & seeds as tuples
+            self.instances = [InstSeedKey(i, s) for s in seeds for i in self.instances]
```

Comment:
This behavior is different than the one in the parent class, could you please make the two look alike?

Reply:
Is it ok if I'll leave this open for now, since Marius had also mentioned we might have to redesign the inheritance of intensifiers?

Reply:
If you create a new issue with open TODOs inherited from this PR, then yes.

Reply:
Okay, I have noted this.